tests/phpunit/Civi/API/RequestTest.php

```diff
@@ -68,4 +68,34 @@ public function testCreateRequest_InvalidEntityAction($inEntity, $inAction, $inV
     Request::create($inEntity, $inAction, ['version' => $inVersion], NULL);
   }
 
+  public function getCastingExamples(): array {
+    $exs = [];
+    // We run the same tests on `$checkPermissions` (which has real PHP setter method)
+    // and `$useTrash` (which has a generic magic method) to show that casting is similar.
+    $exs[] = ['Contact.delete checkPermissions', 0, FALSE];
+    $exs[] = ['Contact.delete checkPermissions', '0', FALSE];
+    $exs[] = ['Contact.delete checkPermissions', 1, TRUE];
+    $exs[] = ['Contact.delete checkPermissions', '1', TRUE];
+    $exs[] = ['Contact.delete useTrash', 0, FALSE];
+    $exs[] = ['Contact.delete useTrash', '0', FALSE];
+    $exs[] = ['Contact.delete useTrash', 1, TRUE];
+    $exs[] = ['Contact.delete useTrash', '1', TRUE];
+    return $exs;
+  }
+
+  /**
+   * @param $entityActionField
+   * @param $inputValue
```

Comment:
Is the type for this `int|string` ?

Reply:
I think it's meant to be mixed, as we're handling how values of various types are handled